xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererShaders.cpp

```diff
@@ -212,8 +212,7 @@ DXGI_FORMAT CRendererShaders::CalcIntermediateTargetFormat(const VideoPicture& p
     return format;
 
   // Preserve HDR precision
-  if (picture.colorBits > 8 && (picture.color_transfer == AVCOL_TRC_SMPTE2084 ||
-                                picture.color_transfer == AVCOL_TRC_ARIB_STD_B67))
+  if (picture.colorBits > 8 && picture.isHdr)
```

Comment:
Before and after are not equivalent (primaries were not taken into account before). I'm OK to use the primaries, but with OR operator: colorbits > 8 && (HDR transfer function or HDR primaries)

Or let me submit instead a PR that uses 10 bits in most situation, except when setting high precision is off and 10 bit SDR is off. (as commented in a previous PR, simply converting YUV 8 bits to RGB 8 bits loses a bit of precision so 10 bits is justified most times)

Reply:
Added that change in a separate commit, the new condition is:

bits > 8 || isHDR (transfer PQ or HLG) || primaries BT.2020